### DISCUSSION TITLE
HIt a gotcha during qsv cat columns command

### DISCUSSION THREAD

**Discussion by reader3** [dathere/qsv#2198] [Q&A] [answered]
I had 3 csv files I was concatenating together, but one file had empty rows, so when the qsv cat columns command executed, it did NOT put the full ranges of rows into the final output file. About a 100,000 rows were missed, but the data was valid. Is there some special option that I should have turned on to make sure that the concatenation would include empty rows, and not prematurely terminate this operation?

**Comment by jqnatividad** [comment]
Thanks for the report @reader3. Can you create a small reproducible example?

**Comment by jqnatividad** [comment, marked as answer]
@reader3 , since you didn't provide a small reproducible example, I created one.

[testcatcolempty.csv](https://github.com/user-attachments/files/17521554/testcatcolempty.csv)
[testcatcol3.csv](https://github.com/user-attachments/files/17521555/testcatcol3.csv)
[testcatcol2.csv](https://github.com/user-attachments/files/17521556/testcatcol2.csv)
[testcatcol1.csv](https://github.com/user-attachments/files/17521557/testcatcol1.csv)

```
$ qsv cat columns testcatcol1.csv testcatcol2.csv testcatcol3.csv | qsv table
col1  col2  col3  col1  col2  col3  col1  col2  col3
2     28    19    2     28    19    2     28    19
6     56    11    6     56    11    6     56    11
51    86    11    51    86    11    51    86    11
39    68    18    39    68    18    39    68    18
85    69    36    85    69    36    85    69    36
```

since testcatcolempty.csv is empty, we need to use the `--pad` option
```
$ qsv cat columns --pad testcatcol1.csv testcatcolempty.csv testcatcol2.csv testcatcol3.csv | qsv table
col1  col2  col3  col1  col2  col3  col1  col2  col3  col1  col2  col3
2     28    19                      2     28    19    2     28    19
6     56    11                      6     56    11    6     56    11
51    86    11                      51    86    11    51    86    11
39    68    18                      39    68    18    39    68    18
85    69    36                      85    69    36    85    69    36
```
